plugins/outputs/azure_data_explorer/azure_data_explorer_test.go

```diff
@@ -0,0 +1,148 @@
+package azure_data_explorer
+
+import (
+	"bufio"
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io"
+	"testing"
+	"time"
+
+	"github.com/Azure/azure-kusto-go/kusto"
+	"github.com/Azure/azure-kusto-go/kusto/ingest"
+	"github.com/influxdata/telegraf"
+	telegrafJson "github.com/influxdata/telegraf/plugins/serializers/json"
+	"github.com/influxdata/telegraf/testutil"
+	"github.com/stretchr/testify/require"
+)
+
+const createTableCommandExpected = `.create-merge table ['%s']  (['fields']:dynamic, ['name']:string, ['tags']:dynamic, ['timestamp']:datetime);`
+const createTableMappingCommandExpected = `.create-or-alter table ['%s'] ingestion json mapping '%s_mapping' '[{"column":"fields", "Properties":{"Path":"$[\'fields\']"}},{"column":"name", "Properties":{"Path":"$[\'name\']"}},{"column":"tags", "Properties":{"Path":"$[\'tags\']"}},{"column":"timestamp", "Properties":{"Path":"$[\'timestamp\']"}}]'`
+
+func TestWrite(t *testing.T) {
+	testCases := []struct {
+		name               string
+		inputMetric        []telegraf.Metric
+		client             *fakeClient
+		createIngestor     ingestorFactory
+		expected           map[string]interface{}
+		expectedWriteError string
+	}{
+		{
+			name:        "Valid metric",
```

Comment:
looks like you could add a test that verifies that fields and tags and timestamps are serialized properly.